modules/programs/starship.nix

```diff
@@ -101,6 +103,7 @@ in {
     programs.fish.interactiveShellInit = mkIf cfg.enableFishIntegration ''
       if test "$TERM" != "dumb"  -a \( -z "$INSIDE_EMACS"  -o "$INSIDE_EMACS" = "vterm" \)
         eval (${starshipCmd} init fish)
+        ${lib.optionals cfg.enableTransience "enable_transience"}
```

Comment:
Have you tested this? I'd expect it to fail since `optionals` produces a list and you're interpolating a string...

Please add a test

Reply:
> Have you tested this? 

Yes, I'm running currently off of my branch, and the screenshot in the PR description is from using it. The bottom half where it has just `>` instead of the colors is the feature. 

```nix
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
    stable.url = "github:NixOS/nixpkgs/22.11";
    #home-manager = { url = "github:nix-community/home-manager"; inputs.nixpkgs.follows = "nixpkgs"; };
    home-manager = { url = "github:will/home-manager/starship-transience"; inputs.nixpkgs.follows = "nixpkgs"; };
```

However I'll switch the conditional to a more suitable one.

> Please add a test

I'm all for adding tests, but haven't yet written any tests in nix. And the other related tests for starship only have tests for the generated toml file, and not for the hooks it puts into shells. Do you have an example of another project that has a test for the `interactiveShellInit` hooks that I can learn from?

Reply:
Right, it only fails when `enableTransience` is `false` since `optionals false ...` yields `[]`. Use `optionalString` instead.